src/utils/oauth2_util.py

```diff
@@ -76,3 +102,44 @@ def encrypt_password(password):
     hash_password = bcrypt.hashpw(password_bytes, salt)
     decode_hash_password = hash_password.decode("utf-8")
     return decode_hash_password
+
+
+def validate_roles_and_scopes(secret_key, endpoint_request):
+
+    if (
+        SecuredHttpRequestUrl(
+            path=endpoint_request.path, method=endpoint_request.method
+        )
+        in roles_endpoint_access
+    ):
+
+        token = endpoint_request.args.get("access_token")
+        token_decode = jwt.decode(token, secret_key, algorithms="HS256")
+        role = token_decode.get("role")
+        scopes = token_decode.get("scopes")
+
+        if (
+            role
+            and role
+            not in roles_endpoint_access[
+                SecuredHttpRequestUrl(
+                    path=endpoint_request.path, method=endpoint_request.method
+                )
+            ]
+        ):
+            raise UnAuthorizedEndpoint("Unauthorized Endpoint Access")
```

Comment:
Looks same error is repeated as well with same message, put it into a variable as well